rust/gelato/src/fwd_req_call.rs

```diff
@@ -0,0 +1,267 @@
+use crate::err::GelatoError;
+use crate::fwd_req_op::ForwardRequestOpArgs;
+use ethers::types::Signature;
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Serialize, Serializer};
+use std::sync::Arc;
+use tracing::info;
+use tracing::instrument;
+
+const GATEWAY_URL: &str = "https://gateway.api.gelato.digital";
+
+#[derive(Debug, Clone)]
+pub struct ForwardRequestCall {
+    pub http: Arc<reqwest::Client>,
+    pub args: ForwardRequestOpArgs,
+    pub sig: Signature,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ForwardRequestCallResult {
+    pub task_id: String,
+}
+
+impl ForwardRequestCall {
+    #[instrument]
+    pub async fn run(
+        &self,
+    ) -> Result<ForwardRequestCallResult, GelatoError> {
+        let url = format!(
+            "{}/metabox-relays/{}",
+            GATEWAY_URL,
+            u32::from(self.args.chain_id)
+        );
+        let http_args = HTTPArgs {
+            args: self.args.clone(),
+            sig: self.sig,
+        };
+        info!(?url, ?http_args);
+        let res = self.http.post(url).json(&http_args).send().await?;
+        let result = HTTPResult::from(res.json().await.unwrap());
+        Ok(ForwardRequestCallResult::from(result))
+    }
+}
+
+#[derive(Debug)]
+struct HTTPArgs {
+    args: ForwardRequestOpArgs,
+    sig: Signature,
+}
+
+#[derive(Debug, Clone, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+struct HTTPResult {
+    pub task_id: String,
+}
+
+impl Serialize for HTTPArgs {
```

Comment:
There were enough modifications I had to make to the serde attribute invocation that it seemed more readable to me to write it out explicitly:

-  hoist the two subtypes up to the top-level dict (`#[serde(flatten)]`)
- make sure the integers for gas and maxfee were enclosed within quotes, since they seem to be javascript bignums (or something)
- add a new top-level dict field 'typeId' with const literal value 'ForwardRequest'
- make sure that the sponsor signature field is prefixed with '0x' since it otherwise just looks like a string

If you still think this would be better and more readable even with all of these customizations to the serde() attribute, I'm open to making the change.

Reply:
Nope, but please more or less copy this to a doc in the code.

Reply:
Done, thanks!